arangod/Cluster/Maintenance.cpp

```diff
@@ -1466,7 +1466,9 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
 
           std::string c = key.substr(pos + 1, pos2);
           std::string s = key.substr(pos2 + 1);  // shard name
-          auto const pdb = pit->second->slice();
+          TRI_ASSERT(pit->second->slice().isArray());
+          TRI_ASSERT(pit->second->slice().length() == 1);
```

Comment:
how can we guarantee that?

Reply:
Only by the way the `plan` argument is created in the first place. Same holds for `current` and `local`. The proper solution IMO would be to replace VelocyPack with static data structures. But that touches the whole maintenance.

So we should introduce tests in this PR for `reportInCurrent`, and the code that writes it input in the first place.

Reply:
@gnusi, yes it is guaranteed. You will not ever get here without a plan object.

Reply:
@kvahed having an array of exactly one element looks really strange. Where is the code building this piece of the plan located?